packages/firestore/src/api/database.ts

```diff
@@ -990,31 +990,24 @@ export class DocumentSnapshot implements firestore.DocumentSnapshot {
     private _fromCache: boolean
   ) {}
 
-  data(): firestore.DocumentData {
+  data(): firestore.DocumentData | undefined {
```

Comment:
I basically picked `undefined` based on a two step process:
- I threw a coin to decide between `undefined` and `null`.
- I then disregarded the coin toss and went with `undefined` since it is the only solution that works with our `get()` call. I wouldn't otherwise know what to return from get() for non-existing documents.

I'm glad you came to the same conclusion.